docs_src/design/adr/0023-North-South-Messaging.md

```diff
@@ -0,0 +1,140 @@
+
+The command message request messages should contain a correlation identifier which would be added to any response so that the 3rd party system would know to associate the responses to the original request.
+
+The device service message command subscription, response publishing and general handling should be implemented in the SDKs versus individually in the device services.
+
+![image](command-msg.png)
+
+### Message Structure
+
+The messages to the command service and those relayed to the device service would mimic their REST alternatives.  The difference is that the HTTP REST requests
+
+- have a header which contain the correlation id. 
+
+- have a path that contains the target device and command name
+
+- are made via HTTP GET or PUT methods signaling get or set actions on the device.
+
+The message bus and the messages on the bus need this additional information. To address this, the command requests must be embedded in a “wrapper message” structure that would contain the additional elements that would be present in the HTTP request messages.
+
+![image](command-msg-structure.png)
+
+Specifically, the message structure would look something like the following:
+
+``` json
+{
+“method” : “PUT”
+“name”: “sensor01”
+“command”:”command01”
+“correlation-id”: “14a42ea6-c394-41c3-8bcd-a29b9f5e6835”
+“request”:
+    {
+        "AHU-TargetTemperature": "28.5",
+        "AHU-TargetBand": "4.0",
+        "AHU-TargetHumidity": {
+        "Accuracy": "0.2-0.3% RH",
+        "Value": 59
+    }
+}
+```
+
+Response messages would be similarly wrapped so as to preserve the correlation id - if nothing else.
+
+``` json
+{
+“correlation-id”: “14a42ea6-c394-41c3-8bcd-a29b9f5e6835”
+“response”:
+    {
+        "apiVersion": "v2",
+        "requestId": "e6e8a2f4-eb14-4649-9e2b-175247911369",
+        "statusCode": 0,
+        "message": "string"
+    }
+}
+```
+
+### Configuration
+
+The configuration or the command service must provide the details in order to subscribe to command messages from any 3rd party system and publish to a response topic.  The subscription configuration would be similar to how core data is configured to subscribe to messages from any device service and publish to topics to get messages to application services.  However, the command service will actually need to have two sets of subscription and two sets of publishing topics.
+
+The command service would subscribe to any command topic (“edgex/request/command/#”) to receive requests from the 3rd party system.  It must then relay these messages (via publish) to the device service actuation topics.
+
+As each device service will have a topic assigned to it, the device service name and device name will be added to the destination topic (“edgex/request/actuate/<device-service>/<device-name>”) when command relays the actuation request messages to device services.
```

Comment:
The character `“` causes some problem when displaying the content, so please try to use ` instead